homeassistant/components/vicare/sensor.py

```diff
@@ -0,0 +1,265 @@
+"""Viessmann ViCare sensor device."""
+import logging
+
+import requests
+
+from homeassistant.const import (
+    CONF_ICON,
+    CONF_NAME,
+    CONF_UNIT_OF_MEASUREMENT,
+    ENERGY_KILO_WATT_HOUR,
+    POWER_WATT,
+    TEMP_CELSIUS,
+    UNIT_PERCENTAGE,
+)
+from homeassistant.helpers.entity import Entity
+
+from . import (
+    DOMAIN as VICARE_DOMAIN,
+    PYVICARE_ERROR,
+    VICARE_API,
+    VICARE_HEATING_TYPE,
+    VICARE_NAME,
+    HeatingType,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_GETTER = "getter"
+
+SENSOR_TYPE_TEMPERATURE = "temperature"
+
+SENSOR_OUTSIDE_TEMPERATURE = "outside_temperature"
+SENSOR_SUPPLY_TEMPERATURE = "supply_temperature"
+SENSOR_RETURN_TEMPERATURE = "return_temperature"
+
+# gas sensors
+SENSOR_BOILER_TEMPERATURE = "boiler_temperature"
+SENSOR_BURNER_MODULATION = "burner_modulation"
+SENSOR_BURNER_STARTS = "burner_starts"
+SENSOR_BURNER_HOURS = "burner_hours"
+SENSOR_BURNER_POWER = "burner_power"
+SENSOR_DHW_GAS_CONSUMPTION_TODAY = "hotwater_gas_consumption_today"
+SENSOR_DHW_GAS_CONSUMPTION_THIS_WEEK = "hotwater_gas_consumption_heating_this_week"
+SENSOR_DHW_GAS_CONSUMPTION_THIS_MONTH = "hotwater_gas_consumption_heating_this_month"
+SENSOR_DHW_GAS_CONSUMPTION_THIS_YEAR = "hotwater_gas_consumption_heating_this_year"
+SENSOR_GAS_CONSUMPTION_TODAY = "gas_consumption_heating_today"
+SENSOR_GAS_CONSUMPTION_THIS_WEEK = "gas_consumption_heating_this_week"
+SENSOR_GAS_CONSUMPTION_THIS_MONTH = "gas_consumption_heating_this_month"
+SENSOR_GAS_CONSUMPTION_THIS_YEAR = "gas_consumption_heating_this_year"
+
+# heatpump sensors
+SENSOR_COMPRESSOR_STARTS = "compressor_starts"
+SENSOR_COMPRESSOR_HOURS = "compressor_hours"
+
+SENSOR_TYPES = {
+    SENSOR_OUTSIDE_TEMPERATURE: {
+        CONF_NAME: "Outside Temperature",
+        CONF_ICON: "mdi:thermometer",
```

Comment:
Please set the temperature device_class and remove the custom icon.

Please review all sensor types and set the appropriate device class when possible.

Here are the available device classes for sensors:
https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

Reply:
Thanks for the suggestion, I've added the device classes, I hope i got them right. 
For example it's not clear to me if a energy meter measured in kWh is also device class power, since it's not explicitly mentioned in the documentation ;)

Reply:
No, those are separate measurements.